OpenSim/Region/CoreModules/Scripting/LSLHttp/UrlModule.cs

```diff
@@ -518,11 +518,16 @@ public void AsyncHttpRequest(IHttpServer server, string path, OSHttpRequest http
                     }
                 }
 
+                // Grab the raw unprocessed original query string, if any.
+                int rawQueryPos = httpRequest.Url.Query.IndexOf('?');
+                string rawQueryStr = (rawQueryPos < 0) ? httpRequest.Url.Query : httpRequest.Url.Query.Substring(rawQueryPos + 1);
+
                 //if this machine is behind DNAT/port forwarding, currently this is being
                 //set to address of port forwarding router
                 requestData.headers["x-remote-ip"] = httpRequest.RemoteIPEndPoint.ToString();
                 requestData.headers["x-path-info"] = pathInfo;
-                requestData.headers["x-query-string"] = queryString;
+                requestData.headers["x-query-string"] = rawQueryStr;        // raw original (SL-compatible)
+                requestData.headers["x-query-string-compat"] = queryString; // processed (old Halcyon scripts)
```

Comment:
I think both of you are thinking the same thing and I definitely agree it may never be used. However, the case I'm worried about is one where someone has handed out a free script that is possibly widely used, by people who are **not** scripters, and can't as Mike said "just fix your script".

Given that this is a possibly breaking change, depending on how the existing query string is used, I just want a backup plan in place that allows us to say to non-scripters "Oh your script doesn't work anymore? Okay just change 'x-query-string' to 'x-query-string-compat' and it's back to it's old behavior."

Without the compat alternative, we're opening ourselves to much more serious complaints after an upgrade, that people who aren't necessarily the authors of scripts are "suddenly forced to _rewrite them_ because we broke the server".  Including a backwards-compatible variant avoids all that.  Given it's dubious likelihood of being used, I wouldn't have included it if it had a high cost, but it's effectively free since it's just returning what the old one did.

Yes, it may not be used, but the InWorldz philosophy is always to do everything possible to avoid breaking content, and if that is required in some fix, provide the safest and least disruptive alternative path.  So in this case, better safe than sorry and provide it as a simple text constant change to fix anything that breaks.